lib/schema.js

```diff
@@ -668,7 +671,20 @@ class Schema {
 	 * @return {Object}
 	 */
 	toJSONSchema() {
-		return this._subschemaToJSONSchema(this.getData());
+		// Used to register JSON Schema definitions at the root of the schema.
+		this.jsonSchemaDefinitions = {};
+
+		let data = this.getData();
+		let jsonSchema = this._subschemaToJSONSchema(data);
+
+		// Attach definitions, if any
+		if (Object.keys(this.jsonSchemaDefinitions).length) {
```

Comment:
Maybe this is just meant to be a virtual method and not actually used, but in that case I would just throw an error saying it needs to be implemented.